libs/python/helperApiTest.py

```diff
@@ -0,0 +1,178 @@
+
+def main():
+    params = read_json_file('/home/user/usecases/inDevelopment/cross_consumption.json')
+    # print(params)
+    data = params["api_resource_uri"]
+    # print(params)
+    # listapi = os.environ['api_resource_uri']
+    # data = json.loads(listapi)
+    for i, d in enumerate(data):
+        flag = True
+        if d["authMethod"] == "basic":
+            del d["oauthurlPathkey"]
+        for k, v in d.items():
+            if d[k] in [None, '', ' ']:
+                logging.error(f"Please provide the value for '{k}' for '{d['servicename']}' service")
+                flag = False
+                break
+        if flag:
+            continue
+        else:
+            break
+    if flag:
+        for i, d in enumerate(data):
+            logging.info("###### API TEST " + str(i) + " #######")
+            logging.info(f"API test for {d['servicename']} service of plan {d['plan']}. Service ID: {d['serviceID']}")
+            if d["APITest"] == "no":
+                continue
+            servicename = d['servicename']
+            plan = d['plan']
+            testapi = d['testapi']
+            if testapi == '/':
+                testapi = ''
+            apiBaseUrlPathKey = d['apiBaseUrlPathKey']
+            clientId_basicUN_PathKey = d['clientId_basicUN_PathKey']
+            clientSecret_basicPW_PathKey = d['clientSecret_basicPW_PathKey']
+            path = f"/home/user/logs/k8s/bindings/{servicename}-{plan}.json"
+            file_exists = exists(path)
+            if not file_exists:
+                continue
+            binding = read_json_file(path)
+            if d['authMethod'] == "oauth":
+                oauthurlPathkey = d['oauthurlPathkey']
+                oauthurl = get_key_values(binding, oauthurlPathkey)
+                uri = get_key_values(binding, apiBaseUrlPathKey)
+                username = get_key_values(binding, clientId_basicUN_PathKey)
+                password = get_key_values(binding, clientSecret_basicPW_PathKey)
+                uri = f"{uri}{testapi}"
+                access_token = get_oauth_token(oauthurl, username, password)
+                get_api_req(uri, access_token)
+            else:
+                uri = get_key_values(binding, apiBaseUrlPathKey)
+                username = get_key_values(binding, clientId_basicUN_PathKey)
+                password = get_key_values(binding, clientSecret_basicPW_PathKey)
+                uri = f"{uri}{testapi}"
+                get_api_req_basic(uri, username, password)
+    file_exists = exists("failedService.txt")
+    if file_exists:
+        file1 = open('failedService.txt', 'r')
+        Lines = file1.readlines()
+        for line in Lines:
+            val = line.split(":")
+            for d in data:
+                if val[0] == d['servicename']:
+                    serviceID = d['serviceID']
+                    logging.error(f"{str(val[0])} service is not crossconsumable. Status : {str(val[1])} ServiceId : {serviceID}")
+
+
+main()
```

Comment:
This method is implicitly called in the `usecase.json` file. Why not make it part of the main flow in the btpsa file?

This would also mean that the main function should be renamed to a "readable" name